plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemPotion.java

```diff
@@ -26,21 +27,41 @@
 import java.util.ArrayList;
 import java.util.List;
 
-// TODO: 1.20.6: Spigot removed the PotionData API, can either remove relevant features on 1.20.6, or try and backsupport somehow (probably either hard-coding our own data or hacking it based on the name)
 public class ItemPotion extends ItemProperty<ObjectTag> {
 
+    // <--[language]
+    // @name Potion Effect Format
+    // @group Minecraft Logic
+    // @description
+    // Potion effects (be it on an entity, given by a potion, etc.) are represented in Denizen as <@link ObjectType MapTag>s with the following keys:
+    // - effect: the effect type given by the effect, see <@link url https://minecraft.wiki/w/Effect#Descriptions>.
+    // - amplifier: the number to increase the effect level by, usually controls how powerful its effects are (optional for input, defaults to 0 which is level 1).
+    // - duration (<@link ObjectType DurationTag>): how long the effect should last (optional for input, defaults to 0s).
+    // - ambient: a boolean (true/false) for whether the effect's particles should be more translucent and less intrusive, like effects given by a beacon (optional for input, defaults to true).
+    // - particles: a boolean (true/false) for whether the effect should display particles (optional for input, defaults to true).
+    // - icon: a boolean (true/false) for whether the effect should have an icon on a player's HUD when applied (optional for input, defaults to false).
+    //
+    // For example, [effect=speed;amplifier=2;duration=10s;ambient=false;particles=true;icon=true] would be a level 3 speed effect that lasts 10 seconds, with (normal) particles and an icon.
+    // -->
+    
     public static boolean describes(ItemTag item) {
         return item.getItemMeta() instanceof PotionMeta || item.getItemMeta() instanceof SuspiciousStewMeta;
     }
 
-    public static MapTag effectToMap(PotionEffect effect) {
+    public static MapTag effectToMap(PotionEffect effect, boolean includeDeprecated) {
         MapTag map = new MapTag();
-        map.putObject("type", new ElementTag(effect.getType().getName()));
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18)) {
+            map.putObject("effect", new ElementTag(Utilities.namespacedKeyToString(effect.getType().getKey()), true));
+        }
         map.putObject("amplifier", new ElementTag(effect.getAmplifier()));
         map.putObject("duration", new DurationTag((long) effect.getDuration()));
         map.putObject("ambient", new ElementTag(effect.isAmbient()));
         map.putObject("particles", new ElementTag(effect.hasParticles()));
         map.putObject("icon", new ElementTag(effect.hasIcon()));
+        // TODO: deprecate this
+        if (includeDeprecated || NMSHandler.getVersion().isAtMost(NMSVersion.v1_19)) {

```

Comment:
I think these version checks are wrong? You have it written such that 1.18 and 1.19 will emit both 'effect' and 'type'

Reply:
Oh yeah forgot to add a note on that - `PotionEffectType` became `Keyed` in 1.18, but the proper registry was only added in 1.20, so wasn't sure whether I should make the `effect` key available where possible or restrict it to match the registry being added.
Was thinking about it from a "why not" perspective & letting scripts work on more versions, but honestly might make more sense to just match it?

Reply:
yeah i'd say just swap at 1.20 to avoid it being weird on some versions (also potentially confusing, you have a key but can't look up by it?)

Reply:
Should be on 1.20+ only now